src/device/cuda/wmma.jl

```diff
@@ -0,0 +1,616 @@
+                                 stride::Number,
+                                 layout::Type{L},
+                                 config::Type{WMMAConfig{M, N, K, T}}) where {T, AS, M, N, K, D_SZ, L}
+
+    as_str                             = get_hl_as_info(AS)
+    layout                             = get_hl_layout(L)
+    shape                              = get_hl_shape(M, N, K)
+    num_els, frag_sz, frag_ty, arr_str = get_hl_frag_info("d", T)
+
+    # Name of the Julia wrapper
+    wrapper = Symbol(join_nonempty("llvm", "wmma", "store", "d", layout, shape, as_str, "stride", arr_str, "_"))
+
+    return quote
+        d_unfl = unflatten(NTuple{$frag_sz, $frag_ty}, d.x)
+        $wrapper(addr, d_unfl, stride)
+        return nothing
+    end
+end
+
+
+# ------------------
+# WMMA fill fragment
+# ------------------
+
+export wmma_fill_c
+
+"""
+    wmma_fill_c(value, config)
+
+Return a [`WMMAFragment`](@ref) filled with the value `value`.
+
+This operation is useful if you want to implement a matrix multiplication (and thus want to set ``C = O``).
+
+# Arguments
+- `value`: The value used to fill the fragment. Can be a `Float16` or `Float32`.
+- `config`: The WMMA configuration that should be used for this WMMA operation. See [`WMMAConfig`](@ref).
+"""
+wmma_fill_c
+
+@generated function wmma_fill_c(value::T,
+                                config::Type{WMMAConfig{M, N, K, D_TYPE}}) where {T, M, N, K, D_TYPE}
+
+    # We can't use closures in @generated functions, so we'll have to do it this way instead of
+    # ntuple(i -> val, $num_els)
+    num_els, _, _ = get_hl_frag_info("c", T)
+
+    args = [:value for i=1:num_els]
+    expr = :(tuple($(args...)))
+
+    return quote
+        return WMMAFragment{$M, $N, $K, $num_els, $T, WMMAUnspecified, WMMAAccumulator}($expr)
+    end
+end
+
+################################################################################
+# BROADCASTING OVER WMMA FRAGMENTS
+################################################################################
+
+# Based on broadcasting implementation of Tuples in
+# https://github.com/JuliaLang/julia/blob/master/base/broadcast.jl
+
+
+# Custom broadcast style for WMMAFragments
+struct WMMAFragmentBroadcastStyle <: Broadcast.BroadcastStyle end
+
+# Use this broadcasting style for WMMAFragments
+Base.BroadcastStyle(::Type{<:WMMAFragment}) = WMMAFragmentBroadcastStyle()
+
+# Broadcast style precedence rules
+# If we broadcast a fragment with a scalar, we want the WMMAFragment style to take precedence
+Base.BroadcastStyle(s::WMMAFragmentBroadcastStyle, t::Broadcast.DefaultArrayStyle{0}) = s
+
+# We don't want to convert fragments before broadcasting
+Base.broadcastable(frag::WMMAFragment) = frag
+
+# Needed for broadcast machinery
+Base.axes(frag::WMMAFragment) = axes(frag.x)
+
+# Helper functions to get element at specified index
+@inline get_index(x, i) = x                           # scalar
+@inline get_index(frag::WMMAFragment, i) = frag.x[i] # WMMAFragment
```

Comment:
Why not plain `getindex` to support `frag[i]`?

Reply:
I require slightly different behaviour for scalars, i.e. `get_index(scalar, i) = scalar` for all `i`, which is why I used a new function `get_index`. But I'll add another `getindex` and then let `get_index` call that to support `frag[i]` too.